### PR TITLE
[PrintingPodRecharge] Fixed JSON file recreation at start time

### PR DESCRIPTION
**Pull request by nightw**
Fixed an error where the bundle JSON files were overwritten at every startup of the game.

Tweaked the logging a bit.

Two other, unrelated small fixes, so the code can compile.

### CODE REVIEW

File: PrintingPodRecharge/DataGen/BundleGen.cs [hunk 1]
```diff
@@ -54,9 +54,10 @@ private static void CreatePack(string folder, string fileName, bool force, Func<
 		{
 			var filePath = Path.Combine(folder, fileName + ".json");
 
-			Log.Debuglog($"Creating pack {filePath}. {File.Exists(filePath)}");
+			Log.Debuglog($"Checking if {filePath} exists: {File.Exists(filePath)}");
 			if (force || !File.Exists(filePath))
 			{
+				Log.Debuglog($"Creating pack {filePath}");
```
Comment (nightw) [inline PrintingPodRecharge/DataGen/BundleGen.cs:60]: This is the small logging tweak

File: PrintingPodRecharge/Mod.cs [hunk 2]
```diff
@@ -81,7 +81,7 @@ private void CreateConfigDirectory()
         public override void OnAllModsLoaded(Harmony harmony, IReadOnlyList<KMod.Mod> mods)
         {
             base.OnAllModsLoaded(harmony, mods);
-            DataGen.BundleGen.Generate(Path.Combine(ModAssets.GetRootPath(), "data", "bundles"), true);
+            DataGen.BundleGen.Generate(Path.Combine(ModAssets.GetRootPath(), "data", "bundles"), false);
```
Comment (nightw) [inline PrintingPodRecharge/Mod.cs:84]: This is the actual fix

File: PrintingPodRecharge/ModAssets.cs [hunk 3]
```diff
@@ -60,11 +60,11 @@ public static void LateLoadAssets()
 
 			Prefabs.bioInkSideScreen = bundle.LoadAsset<GameObject>("Assets/UIs/BioInkSidescreen.prefab");
 			Log.Assert("sidescreen", Prefabs.bioInkSideScreen);
-			tmp.ReplaceAllText(Prefabs.bioInkSideScreen);
+			TMPConverter.ReplaceAllText(Prefabs.bioInkSideScreen);
 
 			Prefabs.settingsDialog = bundle.LoadAsset<GameObject>("Assets/UIs/SettingsDialog 1.prefab");
 			Log.Assert("settingsDialog", Prefabs.settingsDialog);
-			tmp.ReplaceAllText(Prefabs.settingsDialog);
+			TMPConverter.ReplaceAllText(Prefabs.settingsDialog);
```
Comment (nightw) [inline PrintingPodRecharge/ModAssets.cs:67]: This was a compile time error (possibly either .NET or C# version changes caused this to be not compilable anymore), but no meaningful change in functionality. The complaint was technically correct, but I do not know why it was a compile time error, it was CS0176

File: PrintingPodRecharge/Patches/LocalizationPatch.cs [hunk 4]
```diff
@@ -10,7 +10,7 @@ public class Localization_Initialize_Patch
         {
             public static void Postfix()
             {
-                Loc.Translate(typeof(STRINGS), true);
+                Translations.RegisterForTranslation(typeof(STRINGS), true);
             }
```
Comment (nightw) [inline PrintingPodRecharge/Patches/LocalizationPatch.cs:14]: I have absolutely no idea what I am doing here. It just complained that `Loc` does not exist (probably ONI changed the `Localization` class in an update recently), so I applied the same change you made to MoreMarbleSculptures [here](https://github.com/aki-art/ONI-Mods/commit/19a245cdbe2ce5367923a03e2a52a00d7698086f#diff-b6c3810ccf54168f7c52ca907348d8c8b5c5986b4632b531ae7cbeb1678f0a43)

File: PrintingPodRecharge/Mod.cs (hunk 2 above)
Comment (aki-art) [reply PrintingPodRecharge/Mod.cs:84]: If you discovered a bug with the functionality please open an issue so i can look into it. 
The point of putting them outside is to let users modify the packs. By forcing this, you are overwriting the user made jsons, defeating the point entirely. 
If I always wanted to force pack data, i would put them in my mod folder, which gets force updated anyway with each mod updater.

File: PrintingPodRecharge/Patches/LocalizationPatch.cs (hunk 4 above)
Comment (aki-art) [reply PrintingPodRecharge/Patches/LocalizationPatch.cs:14]: Also outdated Futility

**Comment by aki-art** [comment]
Explained in comments, but as summary:
There is no issue i know of to be fixed, while the proposed changes would introduce a problem for sure.

File: PrintingPodRecharge/Mod.cs (hunk 2 above)
Comment (nightw) [reply PrintingPodRecharge/Mod.cs:84]: The actual bug is that in the **current**, live version of the code it is forcing a write overwriting the data, which I fixed in this pull request by changing it to **not** overwrite it at every start. The last function parameter is called `force` and if it is ture then the files will get overwritten even if they exist, see here: https://github.com/aki-art/ONI-Mods/blob/bio-inks/PrintingPodRecharge/DataGen/BundleGen.cs#L35-L58

**Comment by nightw** [comment]
I reacted to the comment, the actual bugfix is valid, the current behavior is the always overwrite one, not my change. Please check my answer inline.

I did not notice that you are using development branches, I can rebase this on the `bio-inks` branch for convenience if that help.

**Pull request closed without merging**